java/runtime/src/main/java/io/ray/runtime/AbstractRayRuntime.java

```diff
@@ -78,6 +78,21 @@ public AbstractRayRuntime(RayConfig rayConfig) {
     return new ObjectRefImpl<T>(objectId, (Class<T>) (obj == null ? Object.class : obj.getClass()));
   }
 
+  @Override
+  public <T> ObjectRef<T> put(T obj, BaseActorHandle actor) {
+    Preconditions.checkNotNull(
+        gcsClient,
+        "Putting an object with assigned actor as owner is not implemented in local mode.");
```

Comment:
It's too weird to use GCSClient to check if it's local mode. 
And I saw the implementation code in LocalModelObjectStore, so let's remove this assertion.

Reply:
This is used to guard the later `gcsClient.getActorAddress(actor.getId())`, in LocalModeObjectStore this would be a null pointer exception.